builder/amazon/common/step_run_source_instance.go

```diff
@@ -275,7 +277,29 @@ func (s *StepRunSourceInstance) Run(state multistep.StateBag) multistep.StepActi
 	if _, exists := s.Tags["Name"]; !exists {
 		s.Tags["Name"] = "Packer Builder"
 	}
-	ec2Tags := ConvertToEC2Tags(s.Tags)
+
+	ec2Tags := make([]*ec2.Tag, 1, len(s.Tags))
+	for k, v := range s.Tags {
+		s.Ctx.Data = &BuildInfoTemplate{
+			SourceAMI:   s.SourceAMI,
+			BuildRegion: *ec2conn.Config.Region,
+		}
+		interpolatedValue, err := interpolate.Render(v, &s.Ctx)
+		if err != nil {
+			err = fmt.Errorf("Error processing tag: %s:%s - %s", k, v, err)
+			state.Put("error", err)
+			ui.Error(err.Error())
+			return multistep.ActionHalt
+		}
+		ec2Tags = append(ec2Tags, &ec2.Tag{Key: aws.String(k), Value: aws.String(interpolatedValue)})
+	}
+	ec2Tags, err = ConvertToEC2Tags(s.Tags, *ec2conn.Config.Region, s.SourceAMI, s.Ctx, ui)
```

Comment:
actually doesn't this overwrite the `ec2Tags` var above? Possibly another rebase artifact